vueui/src/components/view/machinery/vending.vue

```diff
@@ -0,0 +1,181 @@
+<template>
+  <div>
+    <template v-if="mode == 0 || sel_price == 0">
+      <div class="cancel-button">
+        <vui-button v-if="coin" :params="{ remove_coin: 1 }" icon="sign-out-alt">{{ coin }}</vui-button>
+      </div>
+      <div class="t-parent">
+        <vui-button :class="in_stock(vend_item.amount)" class="t-child tooltip" :disabled="vend_item.amount == 0 || mode == 1" push-state :params="{ vendItem: vend_item.key }" v-for="vend_item in products" :key="vend_item.key">
+          <div class="t-container">
+            <vui-img :class="in_stock(vend_item.amount)" class="food-icon" :name="getImage(vend_item)"/>
+            <span v-if="vend_item.price > 0" class="cart-icon fas ic-shopping-cart"/>
+            <span v-if="vend_item.price > 0" class="price">{{ vend_item.price }}电</span>
+            <span class="qty" :class="in_stock(vend_item.amount)">(x{{ vend_item.amount }})</span>
+          </div>
+          <span class="tooltiptext">{{ vend_item.name }}</span>
+        </vui-button>
+      </div>
+    </template>
+    <template v-else-if="sel_name && sel_price > 0">
```

Comment:
It might be worth to add a failsafe v-else with a button to return the vending machine to a known working state. (item overview)